src/components/MovieListPage/MovieListPage.jsx

```diff
@@ -12,68 +13,98 @@ import "../MovieForm/style.css";
-      genres: ["Action & Adventure", "Criminal"],
-    },
-    {
-      id: 2,
-      imageUrl: "./images/Inception.jpg",
-      name: "Inception",
-      releaseYear: 2011,
-      rating: 9.3,
-      duration: "2 hours 46 minutes",
-      description: "Sci-fi movie with Dicaprio",
-      genres: ["Action & Adventure"],
-    },
-    {
-      id: 3,
-      imageUrl: "./images/Kill Bill.jpg",
-      name: "Kill Bill",
-      releaseYear: 1992,
-      rating: 7.1,
-      duration: "1 hours 53 minutes",
-      description: "Another criminal movie by Tarantino",
-      genres: ["Oscar winning Movie"],
-    },
-    {
-      id: 4,
-      imageUrl: "./images/Avengers.jpg",
-      name: "Avengers",
-      releaseYear: 2019,
-      rating: 7.3,
-      duration: "2 hours 26 minutes",
-      description: "Movie by comics",
-      genres: ["Action & Adventure"],
-    },
-  ]);
+  const [querySearch, setQuerySearch] = useState("");
+  const [sortOrder, setSortOrder] = useState("title");
+  const [genre, setGenre] = useState("All");
+  const [moviesInfo, setMoviesInfo] = useState([]);
   const [selectedMovie, setSelectedMovie] = useState(null);
-  const [showModal, setShowModal] = useState(false);
+  const [isFound, setIsFound] = useState(false);
+  const [movieInfo, setMovieInfo] = useState({});
+
+  const genres = [
+    { id: 1, name: "All" },
+    { id: 2, name: "Adventure" },
+    { id: 3, name: "Action" },
+    { id: 4, name: "Science Fiction" },
+    { id: 5, name: "Fantasy" },
+    { id: 6, name: "Family" },
+    { id: 7, name: "Comedy" },
+  ];
+
+  useEffect(() => {
+    console.log("Get all movies");
+    axios
+      .get("http://localhost:4000/movies")
+      .then((response) => {
+        const movies = response.data.data;
+        setMoviesInfo(movies);
+      })
+      .catch((error) => {
+        console.log("Error while getting movies :" + error);
+      });
+    // const movies = getAll();
+    // console.log("from service: " + movies);
+    // setMoviesInfo(movies);
+  }, []);
+
+  useEffect(() => {
+    console.log("Search and filter movies");
+    if (genre === "All") {
+      console.log("genre: " + genre);
+      axios
+        .get("http://localhost:4000/movies", {
+          params: {
+            sortBy: sortOrder,
+            sortOrder: "asc",
+            searchBy: "title",
+            search: querySearch.trim().toLowerCase(),
+          },
+        })
+        .then((response) => {
+          const movies = response.data.data;
+          console.log(movies);
+          setMoviesInfo(movies);
+        });
+    } else {
+      axios
+        .get("http://localhost:4000/movies", {
+          params: {
+            sortBy: sortOrder,
+            sortOrder: "asc",
+            searchBy: "title",
+            search: querySearch.trim().toLowerCase(),
+            filter: genre,
```

Comment:
Let's don't use if...else here. It is better:

`filter: genre !== All && genre`